thor/tests/test_main.py

```diff
@@ -111,3 +114,84 @@ def test_range_and_transform(object_id, orbits, observations):
     # Ensure we get all the object IDs back that we expect
     obs_ids_actual = transformed_detections.id.unique().sort()
     assert pc.all(pc.equal(obs_ids_actual, obs_ids_expected))
+
+
+@pytest.mark.parametrize(
+    "object_id",
+    [
+        pytest.param(OBJECT_IDS[0], marks=pytest.mark.xfail(reason="Fails OD")),
+    ]
+    + OBJECT_IDS[1:3]
+    + [
+        pytest.param(OBJECT_IDS[3], marks=pytest.mark.xfail(reason="Fails OD")),
+        pytest.param(OBJECT_IDS[4], marks=pytest.mark.xfail(reason="Fails OD")),
+        pytest.param(OBJECT_IDS[5], marks=pytest.mark.xfail(reason="Fails OD")),
+    ]
+    + [OBJECT_IDS[6]]
+    + [
+        pytest.param(OBJECT_IDS[7], marks=pytest.mark.xfail(reason="Fails OD")),
+        pytest.param(OBJECT_IDS[8], marks=pytest.mark.xfail(reason="Fails OD")),
+    ]
+    + OBJECT_IDS[9:],
+)
+@pytest.mark.integration
+def test_link_test_orbit(object_id, orbits, observations):
+
+    orbit = orbits.select("object_id", object_id)
+    exposures, detections, associations = observations
+
+    # Select the associations that match this object ID
+    associations_i = associations.select("object_id", object_id)
+    detections_i = detections.apply_mask(
+        pc.is_in(detections.id, associations_i.detection_id)
+    )
+    exposures_i = exposures.apply_mask(pc.is_in(exposures.id, detections_i.exposure_id))
+    assert len(associations_i) == 90
+
+    # Limit detections to first two weeks
+    time_mask = pc.and_(
+        pc.greater_equal(detections_i.time.days, pc.min(detections_i.time.days)),
+        pc.less_equal(
+            detections_i.time.days, pc.min(detections_i.time.days).as_py() + 14
+        ),
+    )
+    detections_i = detections_i.apply_mask(time_mask)
+    exposures_i = exposures_i.apply_mask(
+        pc.is_in(exposures_i.id, detections_i.exposure_id)
+    )
+    associations_i = associations_i.apply_mask(
+        pc.is_in(associations_i.detection_id, detections_i.id)
+    )
+
+    # Extract the observations that match this object ID
+    obs_ids_expected = associations_i.detection_id.unique().sort()
+
+    # Make THOR observations from the detections and exposures
+    observations = Observations.from_detections_and_exposures(detections_i, exposures_i)
+
+    if object_id in TOLERANCES:
+        tolerance = TOLERANCES[object_id]
+    else:
+        tolerance = TOLERANCES["default"]
+
+    # Set a filter to include observations within 1 arcsecond of the predicted position
+    # of the test orbit
+    filters = [TestOrbitRadiusObservationFilter(radius=tolerance)]
+
+    # Create a test orbit for this object
+    test_orbit = THORbit.from_orbits(orbit)
+
+    # Run link_test_orbit and make sure we get the correct observations back
+    for i, results in enumerate(
+        link_test_orbit(test_orbit, observations, filters=filters)
+    ):
+        if i == 4:
```

Comment:
Nice, we could also check for the type(s). Maybe in a next iteration of `link_test_orbit` we can wrap the results in a new class like `StepResultEvent` or something and that will have an enum mapping to the correct stage. Or... not and keep it simple!